### PR TITLE
Chore: Sets up store for teams settings page

### PR DESCRIPTION
**Pull request by nithindavid**
Fixes part of #866

### CODE REVIEW

File: app/javascript/dashboard/store/modules/specs/teamMembers/getters.spec.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { getters } from '../../teamMembers';
+import teamMembers from './fixtures';
+
+describe('#getters', () => {
+  it('getTeamMembers', () => {
+    const state = {
+      records: {
+        1: teamMembers[0],
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/specs/teamMembers/getters.spec.js:8]: This is supposed to be an array right.

```js
{
  1: [{ ...agent1 }, { ..agent2 }]
}
```

File: app/javascript/dashboard/store/modules/specs/teamMembers/getters.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  });
+
+  it('getUIFlags', () => {
+    const state = {
+      uiFlags: {
+        isFetching: true,
+        isFetchingItem: false,
+        isCreating: false,
+        isUpdating: false,
+        isUpdatingAutoAssignment: false,
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/specs/teamMembers/getters.spec.js:21]: Please remove irrelevant flags.

File: app/javascript/dashboard/store/modules/specs/teams/getters.spec.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  it('getUIFlags', () => {
+    const state = {
+      uiFlags: {
+        isFetching: true,
+        isFetchingItem: false,
+        isCreating: false,
+        isUpdating: false,
+        isUpdatingAutoAssignment: false,
+        isDeleting: false,
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/specs/teams/getters.spec.js:32]: Please remove unnecessary flags.

File: app/javascript/dashboard/store/modules/teams/mutations.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      ...data,
+    });
+  },
+
+  [EDIT_TEAM]: ($state, data) => {
+    Vue.set($state.records, data.id, data);
+  },
+
+  [DELETE_TEAM]: ($state, teamId) => {
+    const { [teamId]: toDelete, ...records } = $state.records;
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/teams/mutations.js:44]: Where is `toDelete` defined?

File: app/javascript/dashboard/store/modules/specs/teams/mutations.spec.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      mutations[EDIT_TEAM](state, {
+        id: 1,
+        name: 'customer-support',
+      });
+      expect(state.records[1].name).toEqual('customer-support');
+    });
+  });
+
+  describe('#DELETE_TEAM', () => {
+    it('delete label record', () => {
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/specs/teams/mutations.spec.js:39]: label?

File: app/javascript/dashboard/store/modules/specs/teams/mutations.spec.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  describe('#ADD_TEAM', () => {
+    it('push newly created label to the store', () => {
+      const state = { records: {} };
+      mutations[SET_TEAM_ITEM](state, teams[1]);
+      expect(state.records).toEqual({ 1: teams[1] });
+    });
+  });
+
+  describe('#EDIT_TEAM', () => {
+    it('update label record', () => {
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/specs/teams/mutations.spec.js:28]: label?

File: app/javascript/dashboard/store/modules/specs/teams/mutations.spec.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  describe('#SET_teams', () => {
+    it('set label records', () => {
+      const state = { records: {} };
+      mutations[SET_TEAMS](state, [teams[1]]);
+      expect(state.records).toEqual(teams);
+    });
+  });
+
+  describe('#ADD_TEAM', () => {
+    it('push newly created label to the store', () => {
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/specs/teams/mutations.spec.js:20]: label?

File: app/javascript/dashboard/store/modules/specs/teams/mutations.spec.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  SET_TEAMS,
+  SET_TEAM_ITEM,
+  EDIT_TEAM,
+  DELETE_TEAM,
+} from '../../teams/types';
+import { mutations } from '../../teams/mutations';
+import teams from './fixtures';
+describe('#mutations', () => {
+  describe('#SET_teams', () => {
+    it('set label records', () => {
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/specs/teams/mutations.spec.js:12]: label?

File: app/javascript/dashboard/store/modules/specs/teams/getters.spec.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      },
+    };
+    expect(getters.getUIFlags(state)).toEqual({
+      isFetching: true,
+      isFetchingItem: false,
+      isCreating: false,
+      isUpdating: false,
+      isUpdatingAutoAssignment: false,
+      isDeleting: false,
+    });
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/specs/teams/getters.spec.js:42]: please remove unnecessary lines.

File: app/javascript/dashboard/store/modules/teams/actions.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+import TeamsAPI from '../../../api/teams';
+
+export const actions = {
+  create: async ({ commit }, teamInfo) => {
+    commit(SET_TEAM_UI_FLAG, { isCreating: true });
+    try {
+      const response = await TeamsAPI.create(teamInfo);
+      const team = response.data;
+      commit(SET_TEAM_ITEM, team);
+      return team;
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/teams/actions.js:18]: Why is a return required here?

File: app/javascript/dashboard/store/modules/teams/getters.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+export const getters = {
+  getTeams($state) {
+    return Object.values($state.records);
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/teams/getters.js:3]: Would this maintain order? Should we sort based on team id.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please fix the comments.

File: app/javascript/dashboard/store/modules/teams/mutations.js (hunk 4 above)
Comment (nithindavid) [reply app/javascript/dashboard/store/modules/teams/mutations.js:44]: Es6 syntax

File: app/javascript/dashboard/store/modules/teams/actions.js (hunk 10 above)
Comment (nithindavid) [reply app/javascript/dashboard/store/modules/teams/actions.js:18]: To avoid watch in vue

**Comment by nithindavid** [comment]
@pranavrajs Fixed errors

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**